### PR TITLE
feat(dashboard): Migrate dashboard orderbys to function format

### PR DESCRIPTION
**Pull request by shruthilayaj**
They are currently in alias for tables and legacy widgets.
Updating them to store as function names to match the timeseries
widgets built on the new widget builder.

Tested with a few orderbys:
Before
```
In [2]: for query in DashboardWidgetQuery.objects.all():
   ...:     if query.orderby:
   ...:         print(query.orderby)
   ...: 
-count
date
-sum(session)
-sum(session)
-sum(session)
-sum(session)
-count_miserable_user_300
-last_seen
-count_unique_device_charging
```

After
```
In [2]: for query in DashboardWidgetQuery.objects.all():
   ...:     if query.orderby:
   ...:         print(query.orderby)
   ...: 
date
-sum(session)
-sum(session)
-sum(session)
-sum(session)
-count()
-last_seen()
-count_miserable(user,300)
-count_unique(device.charging)
```

Requires: https://github.com/getsentry/sentry/pull/34181

### CODE REVIEW

File: src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    arguments = parse_arguments(match[1], match[2])
+    alias = match[1] + "_" + "_".join(arguments)
+
+    for pattern in [r"[^\w]", r"^_+", r"_+$"]:
+        alias = re.sub(pattern, "_", alias)
+
+    return alias
+
+
+def parse_arguments(function: str, columns: str):
```
Comment (shruthilayaj) [inline src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py:31]: copy pasted from here https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/fields.py#L435

File: src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+AGGREGATE_PATTERN = r"^(\w+)\((.*)?\)$"
+
+
+def get_function_alias(field: str):
```
Comment (shruthilayaj) [inline src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py:13]: Replicating logic from https://github.com/getsentry/sentry/blob/master/static/app/utils/discover/fields.tsx#L1038

File: src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
```
Comment (shruthilayaj) [inline src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py:119]: ~70,000 rows

File: src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        if query.orderby == "":
+            continue
+
+        orderby = query.orderby
+        stripped_orderby = orderby.lstrip("-")
+        orderby_prefix = "-" if orderby.startswith("-") else ""
+
+        for aggregate in query.aggregates:
+            alias = get_function_alias(aggregate)
+            if alias == stripped_orderby:
```
Comment (wmak) [inline src/sentry/migrations/0289_dashboardwidgetquery_convert_orderby_to_field.py:102]: To confirm, I know queries support >1 orderby, but I see `DashboardWidgetQuery` has orderby as a string, this is cause `DashboardWidgetQuery` only supports exactly 1 orderby right?

**Review by wmak — APPROVED** [review]
LGTM, only one clarifying question for my own understanding

**Review by wedamija — COMMENTED** [review]
Would it be valuable to write a test case for this migration?

If you want to do so, you can inherit from `TestMigrations`. There are a few example testcases around already.

**Review by wedamija — APPROVED** [review]

**Pull request merged**